packages/core/src/main/notification/NotificationService.ts

```diff
@@ -257,25 +257,31 @@ export class NotificationService extends EventEmitter {
         const groupIdStr = Encoder.toBase64(newGroupId).asString;
         // The groupId can then be sent back to the consumer
         return {
-          event,
+          event: {...event, data: groupIdStr},
           mappedEvent: ConversationMapper.mapConversationEvent({...event, data: groupIdStr}, source),
         };
 
       case Events.CONVERSATION_EVENT.MLS_MESSAGE_ADD:
         const encryptedData = Decoder.fromBase64(event.data).asBytes;
 
-        const groupId = await this.getUint8ArrayFromConversationGroupId(
+        const groupId = await this.getGroupIdFromConversationId(
           event.qualified_conversation ?? {id: event.conversation, domain: ''},
         );
+        const groupIdBytes = Decoder.fromBase64(groupId).asBytes;
 
         // Check if the message includes proposals
-        const {proposals, commitDelay, message} = await this.mlsService.decryptMessage(groupId, encryptedData);
-        if (proposals.length > 0) {
+        const res = await this.mlsService.decryptMessage(groupIdBytes, encryptedData);
+        const {proposals, commitDelay, message} = res;
```

Comment:
Not a big thing, but why we can't destructure it right away? Is it for readability? I'd suggest some better naming for `res` then, like `decryptedMessage` 😄

Reply:
Oh totally right, this was for debugging only 😱 !!!! Nice catch 👌